includes/class-wp-job-manager-recaptcha.php

```diff
@@ -0,0 +1,231 @@
+<?php
+/**
+ * File containing the WP_Job_Manager_Recaptcha class.
+ *
+ * @package wp-job-manager
+ */
+
+namespace WP_Job_Manager;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * WP_Job_Manager_Recaptcha class.
+ *
+ * @since $$next-version$$
+ * @internal
+ */
+class WP_Job_Manager_Recaptcha {
+
+	use Singleton;
+
+	/**
+	 * Site key.
+	 *
+	 * @var string
+	 */
+	private $site_key;
+
+	/**
+	 * Secret key.
+	 *
+	 * @var string
+	 */
+	private $secret_key;
+
+	/**
+	 * The reCAPTCHA version.
+	 *
+	 * @var string
+	 */
+	private $recaptcha_version;
+
+	const RECAPTCHA_SITE_KEY   = 'job_manager_recaptcha_site_key';
+	const RECAPTCHA_SECRET_KEY = 'job_manager_recaptcha_secret_key';
+	const RECAPTCHA_VERSION    = 'job_manager_recaptcha_version';
+
+	/**
+	 * Initialize class for landing pages.
+	 *
+	 * @since 2.0.0
+	 */
+	public function __construct() {
+		$this->site_key          = get_option( self::RECAPTCHA_SITE_KEY );
+		$this->secret_key        = get_option( self::RECAPTCHA_SECRET_KEY );
+		$this->recaptcha_version = get_option( self::RECAPTCHA_VERSION, 'v2' );
+
+		if ( $this->use_recaptcha_field() ) {
+			add_action( 'submit_job_form_end', [ $this, 'display_recaptcha_field' ] );
+			add_filter( 'submit_job_form_validate_fields', [ $this, 'validate_recaptcha_field' ] );
+			add_filter( 'submit_draft_job_form_validate_fields', [ $this, 'validate_recaptcha_field' ] );
+		}
+	}
+
+	/**
+	 * Use reCAPTCHA field on the form?
+	 *
+	 * @return bool
+	 */
+	public function use_recaptcha_field() {
+		if ( ! $this->is_recaptcha_available() ) {
+			return false;
+		}
+		return 1 === absint( get_option( 'job_manager_enable_recaptcha_job_submission' ) );
+	}
+
+	/**
+	 * Enqueue the scripts for the form.
+	 */
+	public static function enqueue_scripts() {
+		$instance = self::instance();
+
+		if (
+			$instance->use_recaptcha_field() &&
+			in_array( $instance->recaptcha_version, [ 'v2', 'v3' ], true )
+		) {
+			$recaptcha_version = $instance->recaptcha_version;
+			$recaptcha_url     = '';
+
+			if ( 'v2' === $recaptcha_version ) {
```

Comment:
When the user upgrades, the recaptcha_version variable is not set which means that although the had a recaptcha before it will stop working.

I think we should either do a mini migration and set to v2 for such cases or assume v2 when it's empty.

Reply:
I think it should work because we set `v2` as the default here:
`$this->recaptcha_version = get_option( self::RECAPTCHA_VERSION, 'v2' );`
https://github.com/Automattic/WP-Job-Manager/pull/2773/files#diff-20f5f5d09f20927ac0c7a20add75346d466aca6f66eb2ae09daeb54c27c2b785R57

What I tested was deleting the recaptcha version option, setting recaptcha in `trunk`, then moving to the new branch, and it did work for Resumes, Applications, and Core.  

Let me know if I'm missing something else 😄

Reply:
Ah yes indeed the option was set from previous set and this is the reason it didn't work. 

But there is another case, and it doesn't work only for core. If there are no attributes set in the shortcode, `atts` is not going to be an array `abstract-wp-job-manager-form.php` so the recaptcha doesn't work.

Reply:
Thanks for the details!  I've removed the check for `atts` so I think this should work now :)